### PR TITLE
Flow Blueprints: Select a flow blueprint during instance creation

### PR DESCRIPTION
**Pull request by Pezmc**
## Description

Screenshot 2023-10-11 at 09 46 35

## Related Issue(s)

For https://github.com/FlowFuse/flowfuse/issues/2874, this is the quick win, to be followed up with UI improvements down the line.

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass - **Draft as EE UI coverage is needed**
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by knolleary** [comment]
I think the 'Blank Template' should be a hardcoded entry in the front-end, rather than rely on it having been created first in the database.

File: frontend/src/pages/instance/components/InstanceForm.vue [hunk 1]
```diff
@@ -204,11 +205,41 @@
                         v-model="copyParts"
                     />
                 </template>
+                <div v-else-if="creatingNew && flowBlueprintsEnabled && flowBlueprints.length > 1">
```
Comment (knolleary) [inline frontend/src/pages/instance/components/InstanceForm.vue:208]:
```suggestion
                <div v-else-if="creatingNew && flowBlueprintsEnabled && flowBlueprints.length > 0">
```

Think this was an off-by-one error. I only had one legit blueprint in my database, so nothing was being shown.

File: frontend/src/pages/instance/Settings/ChangeInstanceType.vue [hunk 2]
```diff
@@ -1,7 +1,13 @@
 <template>
     <div class="w-full max-w-4xl" data-el="change-project">
         <ff-loading v-if="saving" message="Updating Instance..." />
-        <InstanceForm v-else :instance="instanceDetails || instance" :team="team" :billingEnabled="!!features.billing" @on-submit="changeInstanceDefinition" />
+        <InstanceForm
+            v-else :instance="instanceDetails || instance"
+            :team="team"
+            :billing-enabled="!!features.billing"
+            :flow-blueprints-enabled="!!features.flowBlueprints"
```
Comment (knolleary) [inline frontend/src/pages/instance/Settings/ChangeInstanceType.vue:8]: I don't think in this context (changing an instance type) we ever want to offer the blueprints. I see that is guarded against inside InstanceForm, but would it be more explicit to hardcode this to false here?
```suggestion
            :flow-blueprints-enabled="false"
```

**Review by knolleary — CHANGES_REQUESTED** [review]

File: frontend/src/api/flowBlueprints.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ * @param {number} limit
+ * @returns [{ id:number, active:boolean, name:string, description:string, createdAt:string, updatedAt:string }]
+ */
+const getFlowBlueprints = async (state = 'active', cursor, limit) => {
+    const url = paginateUrl('/api/v1/flow-templates', cursor, limit, null, { filter: state })
+    return client.get(url).then(res => {
+        return res.data.templates
+    })
+}
+export default {
```
[the remark this answers is not in the document]
Comment (Pezmc) [reply frontend/src/api/flowBlueprints.js:18]: I don't think we need this, I already had markdown support, note the fullstops and italics in the screenshot.

**Comment by Pezmc** [comment]
Test coverage will be added as a follow up, marking ready to review so this can go to staging!

**Review by knolleary — APPROVED** [review]

**Pull request merged**